metron-platform/metron-common/src/test/java/org/apache/metron/common/dsl/functions/StringFunctionsTest.java

```diff
@@ -207,4 +207,78 @@ public void testFormatWithNoArguments() throws Exception {
   public void testFormatWithMissingArguments() throws Exception {
     run("FORMAT('missing arg: %d')", Collections.emptyMap());
   }
+
+
+  /**
+  * CHOP StringFunction
+  * @throws Exception
+  */
+  @Test
+  public void testChop() throws Exception {
+    Assert.assertEquals("ab",   run("CHOP('abc')", new HashedMap()));
+    Assert.assertEquals(null,   run("CHOP(null)", new HashedMap()));
+    Assert.assertEquals("abc",  run("CHOP(msg)", ImmutableMap.of("msg", "abc\r\n")));
+    Assert.assertEquals("",     run("CHOP(msg)", ImmutableMap.of("msg", "\n")));
+  }
+
+  @Test(expected = ParseException.class)
+  public void testChopWithMissingArguments() throws Exception {
+    run("CHOP()", Collections.emptyMap());
+  }
+
+  /**
+   * PREPENDIFMISSING StringFunction
+   * @throws Exception
+   */
+  @Test
+  public void testPrependIfMissing() throws Exception {
+    Assert.assertEquals("xyzabc",     run("PREPENDIFMISSING('abc', 'xyz')", new HashedMap()));
+    Assert.assertEquals("xyzXYZabc",  run("PREPENDIFMISSING('XYZabc', 'xyz', 'mno')", new HashedMap()));
+    Assert.assertEquals(null,         run("PREPENDIFMISSING(null, null, null)", new HashedMap()));
+    Assert.assertEquals("xyz",        run("PREPENDIFMISSING('', 'xyz', null)", new HashedMap()));
+  }
+
+  @Test(expected = ParseException.class)
+  public void testPrependIfMissingWithIncorrectArgs() throws Exception {
```

Comment:
I may be wrong, but I think JUnit expected errors do not auto-continue inside an errored test case.  So, you're only testing the first sub-case.  

To fix, you either need 3 test cases, or you need to do try/catch/finally contexts around each `run()`, where the catch eats the exception but sets a marker flag, and the finally asserts the marker flag.